src/cpu/aarch64/acl_gemm_convolution.hpp

```diff
@@ -17,51 +17,24 @@
 #ifndef CPU_AARCH64_ACL_GEMM_CONVOLUTION_HPP
 #define CPU_AARCH64_ACL_GEMM_CONVOLUTION_HPP
 
+#include "common/memory_tracking.hpp"
 #include "cpu/cpu_convolution_pd.hpp"
 
-#include "cpu/aarch64/acl_convolution_utils.hpp"
-#include "cpu/aarch64/acl_post_ops.hpp"
+#include "acl_convolution_utils.hpp"
+#include "acl_post_ops.hpp"
+#include "arm_compute/runtime/experimental/operators/CpuGemmConv2d.h"
 
 namespace dnnl {
 namespace impl {
 namespace cpu {
 namespace aarch64 {
 
-struct acl_resource_t : public resource_t {
-    acl_resource_t()
-        : acl_obj_(utils::make_unique<
-                acl_obj_t<arm_compute::NEGEMMConvolutionLayer>>()) {}
-
-    status_t configure(const acl_conv_conf_t &acp) {
-        if (!acl_obj_) return status::out_of_memory;
-
-        // Init Compute Library tensors based on info from descriptor
-        acl_obj_->src_tensor.allocator()->init(acp.src_tensor_info);
-        acl_obj_->wei_tensor.allocator()->init(acp.wei_tensor_info);
-        acl_obj_->dst_tensor.allocator()->init(acp.dst_tensor_info);
-        acl_obj_->bia_tensor.allocator()->init(acp.bia_tensor_info);
-
-        acl_obj_->conv.configure(&acl_obj_->src_tensor, &acl_obj_->wei_tensor,
-                acp.with_bias ? &acl_obj_->bia_tensor : nullptr,
-                &acl_obj_->dst_tensor, acp.padstride_info, acp.weights_info,
-                acp.dilation_info, acp.act_info, acp.fast_math);
-        return status::success;
-    }
-
-    acl_obj_t<arm_compute::NEGEMMConvolutionLayer> &get_acl_obj() const {
-        return *acl_obj_;
-    }
-
-    DNNL_DISALLOW_COPY_AND_ASSIGN(acl_resource_t);
-
-private:
-    std::unique_ptr<acl_obj_t<arm_compute::NEGEMMConvolutionLayer>> acl_obj_;
-
-}; // acl_resource_t
-
 template <data_type_t src_type, data_type_t wei_type = src_type,
         data_type_t dst_type = src_type, data_type_t bia_type = dst_type>
 struct acl_gemm_convolution_fwd_t : public primitive_t {
+
+    using Op = arm_compute::experimental::op::CpuGemmConv2d;
```

Comment:
The full name is too long, but Op is too ambiguous. Something like ACLGemmConvOp would be clearer

Reply:
There is a lot of infrastructure already in place around this and we have some time pressure to deliver, so keeping this in. At the end of all planned submissions, we can do one final clean up commit addressing these concerns.